src/fmu/ensemble/ensemble.py

```diff
@@ -839,6 +855,26 @@ def drop(self, localpath, **kwargs):
             except ValueError:
                 pass  # Allow localpath to be missing in some realizations
 
+    def process_batch(self, batch=None):
+        """Process a list of functions to run/apply
+
+        This is equivalent to calling each function individually
+        but this enables more efficient concurrency. It is meant
+        to be used for functions that modifies the realization
+        object, not for functions that returns a dataframe already.
+
+        Args:
+            batch (list): Each list element is a dictionary with one key,
+                being a function names, value pr key is a dict with keyword
+                arguments to be supplied to each function.
+        Returns:
+            ScratchEnsemble: This ensemble object (self), for it
+                to be picked up by ProcessPoolExecutor and pickling.
+        """
+        for realidx, realization in self._realizations.items():
```

Comment:
use `.values()`

Reply:
That would not give me the realization index.

Reply:
But you are right, I am not using the index...